ibm/resource_ibm_cse_instance.go

```diff
@@ -0,0 +1,388 @@
+			},
+			"udp_ports": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "The exposed udp ports for the CSE.",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Set:         HashInt,
+			},
+			"tcp_range": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The exposed tcp port range for the CSE.",
+				Optional:    true,
+			},
+			"udp_range": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The exposed udp port range for the CSE.",
+				Optional:    true,
+			},
+			"max_speed": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The maxium network speed for the CSE.",
+				Optional:    true,
+			},
+			"estado_proto": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The estado protocol for the CSE, value could be tcp, http or https.",
+				Optional:    true,
+			},
+			"estado_port": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The estado port for the CSE",
+				Optional:    true,
+			},
+			"estado_path": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The estado path for the CSE",
+				Optional:    true,
+			},
+			"dedicated": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The value is 1 will create the CSE in dedicated device, is 0 will create in shared device.",
+				Optional:    true,
+			},
+			"multi_tenant": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The dedicated device will only run this CSE when the value is 1.",
+				Optional:    true,
+			},
+			"acl": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "The acl setting for this CSE.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"static_addresses": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCSEInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	payload, err := genCreatePayload(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("resourceCSEInstanceCreate: payload=%v\n", payload)
+
+	srvId, err := seAPI.CreateServiceEndpoint(payload)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(srvId)
+	d.Set("service_id", srvId)
+
+	return nil
+}
+
+func resourceCSEInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	srvId := d.Get("service_id").(string)
```

Comment:
Don't use d.Get("...") in read method.. 
srvId := d.Id()----can be used to get srvId

Reply:
Agree with you. Will change it asap.